lxd/api_cluster.go

```diff
@@ -329,6 +447,117 @@ func clusterPutDisable(d *Daemon) Response {
 	return EmptySyncResponse
 }
 
+// Initialize storage pools and networks on this node.
+//
+// We pass to LXD client instances, one connected to ourselves (the joining
+// node) and one connected to the target cluster node to join.
+func clusterInitMember(d, client lxd.ContainerServer, memberConfig []api.ClusterMemberConfigKey) error {
+	data := initDataNode{}
+
+	// Fetch all pools current defined in the cluster.
+	pools, err := client.GetStoragePools()
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch information about cluster storage pools")
+	}
+
+	// Merge the returned storage pools configs with the node-specific
+	// configs provided by the user.
+	for _, pool := range pools {
+		// Skip pending pools.
+		if pool.Status == "PENDING" {
+			continue
+		}
+
+		// Skip ceph pools since they have no node-specific key and
+		// don't need to be defined on joining nodes.
+		if pool.Driver == "ceph" {
+			continue
+		}
+
+		logger.Debugf("populating init data for storage pool %s", pool.Name)
+
+		post := api.StoragePoolsPost{
+			StoragePoolPut: pool.StoragePoolPut,
+			Driver:         pool.Driver,
+			Name:           pool.Name,
+		}
+
+		// Delete config keys that are automatically populated by LXD
+		delete(post.Config, "volatile.initial_source")
+		delete(post.Config, "zfs.pool_name")
+
+		// Apply the node-specific config supplied by the user.
+		for _, config := range memberConfig {
+			if config.Entity != "storage-pool" {
+				continue
+			}
+
+			if config.Name != pool.Name {
+				continue
+			}
+
+			if !shared.StringInSlice(config.Key, db.StoragePoolNodeConfigKeys) {
+				logger.Warnf("Ignoring config key %s for storage pool %s", config.Key, config.Name)
```

Comment:
Log statements should follow same rules as errors

Reply:
Fixed.